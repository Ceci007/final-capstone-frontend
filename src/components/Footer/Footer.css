@import "https://fonts.googleapis.com/css2?family=Roboto&display=swap";

body {
  margin: 0;
  background-color: #f7f8f9;
}

.menu {
  flex-shrink: 0;
  font-family: 'Roboto', sans-serif;
  position: fixed;
  bottom: 0;
  width: 100%;
  height: 10vh;
  display: flex;
  justify-content: space-around;
  background-color: #35434c;
  color: #f7f8f9;
  align-items: center;
}

.menu a {
  color: white;
  text-decoration: none;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 25%;
  height: 100%;
}

.icons {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  cursor: pointer;
}

.icon-btn {
  width: 25%;
}

.menu a:hover,
.icon-btn:hover {
  background-color: #62b5e5;
}

.icons button {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  background: none;
  border: none;
  cursor: pointer;
}

.icons i,
.icons button i {
  color: #f7f8f9;
  font-size: 4.5vh;
}

.icons p,
.icons button p {
  margin: 0;
  color: #f7f8f9;
  text-decoration-line: none;
  font-size: 9px;
  width: 55px;
  width: 40px;
  text-align: center;
}

.add-trackings {
  width: 100% !important;
}
