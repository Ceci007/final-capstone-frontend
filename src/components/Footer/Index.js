import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import './Footer.css';
import PathComponent from '../PathComponent';
import { logOutUser } from '../../actions/user';

const Footer = ({
  isLogin, logOut, history, displayForm, match,
}) => {
  const handleClick = async e => {
    e.preventDefault();
    const response = await logOut();
    if (response.data.logged_out) {
      history.push('/');
    }
  };

  const displayAddForm = () => {
    displayForm();
  };

  return (
    <nav className="menu">
      <PathComponent path="main" icon="fa-bar-chart" linkText="Illnesses" />

      {isLogin && match.path === '/illness/:id' ? (
        <div className="icons icon-btn">
          <button type="button" onClick={displayAddForm}>
            <i className="fa fa-line-chart" />
            <p className="add-trackings">Add Trackings</p>
          </button>
        </div>
      ) : null}

      { !isLogin
        && (
        <>
          <PathComponent path="login" icon="fa-sign-in" linkText="Log In" />
          <PathComponent path="signup" icon="fa-user-circle" linkText="Sign Up" />
        </>
        )}
      { isLogin && (
      <PathComponent path="logout" icon="fa-sign-in" linkText="Log Out" handleClick={e => handleClick(e)} />
      )}
    </nav>

  );
};
Footer.propTypes = {
  isLogin: PropTypes.bool,
  logOut: PropTypes.func,
  history: PropTypes.shape({
    push: PropTypes.func,
  }),

  match: PropTypes.shape({
    path: PropTypes.string,
  }),
  displayForm: PropTypes.func,
};

Footer.defaultProps = {
  isLogin: false,
  logOut: () => {},
  history: {},
  match: {},
  displayForm: () => {},

};

const mapStateToProps = state => ({
  isLogin: state.user.isLogin,
  tracking: state.tracking,

});
const mapDispatchToProps = dispatch => ({
  logOut: () => dispatch(logOutUser()),
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Footer));
