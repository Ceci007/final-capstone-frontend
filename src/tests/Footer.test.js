import thunk from 'redux-thunk';
import { Provider } from 'react-redux'
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import combinedReducers from '../reducers';
import Footer from '../components/Footer/Footer';

test('Footer renders correctly', () => {
  const store = createStore(combinedReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
