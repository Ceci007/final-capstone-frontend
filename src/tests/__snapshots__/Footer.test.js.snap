// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<nav
  className="menu"
>
  <a
    href="/main"
    onClick={[Function]}
  >
    <div
      className="icons"
    >
      <i
        className="fa fa-bar-chart"
      />
      <p>
        Illnesses
      </p>
    </div>
  </a>
  <a
    href="/login"
    onClick={[Function]}
  >
    <div
      className="icons"
    >
      <i
        className="fa fa-sign-in"
      />
      <p>
        Log In
      </p>
    </div>
  </a>
  <a
    href="/signup"
    onClick={[Function]}
  >
    <div
      className="icons"
    >
      <i
        className="fa fa-user-circle"
      />
      <p>
        Sign Up
      </p>
    </div>
  </a>
</nav>
`;
